--- a/SPECS/httpd.spec	2025-06-26 12:42:55
+++ b/SPECS/httpd.spec	2025-06-30 11:29:52
@@ -107,6 +107,9 @@
 
 # Security fixes
 # Patch200: ...
+
+Patch900: mod_proxy_http.patch
+Patch901: proxypass_nomain_flag.patch
 
 # Apache-2.0: everything
 # BSD-3-Clause: util_pcre.c, ap_regex.h
@@ -255,6 +258,7 @@
 %{gpgverify} --keyring='%{SOURCE2}' --signature='%{SOURCE1}' --data='%{SOURCE0}'
 %autosetup -p1 -S gendiff
 
+
 # Patch in the vendor string
 sed -i '/^#define PLATFORM/s/Unix/%{vstring}/' os/unix/os.h
 sed -i 's/@RELEASE@/%{release}/' server/core.c
@@ -542,13 +546,6 @@
 install -m 644 -p $RPM_SOURCE_DIR/httpd.logrotate \
         $RPM_BUILD_ROOT/etc/logrotate.d/httpd
 
-# Install man pages
-install -d $RPM_BUILD_ROOT%{_mandir}/man8 $RPM_BUILD_ROOT%{_mandir}/man5
-install -m 644 -p httpd.service.8 httpd-init.service.8 httpd.socket.8 \
-        httpd@.service.8 htcacheclean.service.8 apachectl.8 \
-        $RPM_BUILD_ROOT%{_mandir}/man8
-install -m 644 -p httpd.conf.5 \
-        $RPM_BUILD_ROOT%{_mandir}/man5
 
 # fix man page paths
 sed -e "s|/usr/local/apache2/conf/httpd.conf|/etc/httpd/conf/httpd.conf|" \
@@ -670,8 +667,6 @@
 
 %files
 %{_mandir}/man8/*
-%{_mandir}/man5/*
-%exclude %{_mandir}/man8/httpd-init.*
 
 %config(noreplace) %{_sysconfdir}/httpd/conf.modules.d/00-brotli.conf
 %config(noreplace) %{_sysconfdir}/httpd/conf.modules.d/00-systemd.conf
@@ -793,7 +788,6 @@
 %{_libexecdir}/httpd-ssl-pass-dialog
 %{_libexecdir}/httpd-ssl-gencerts
 %{_unitdir}/httpd.socket.d/10-listen443.conf
-%{_mandir}/man8/httpd-init.*
 
 %files -n mod_proxy_html
 %{_libdir}/httpd/modules/mod_proxy_html.so
