--- a/SPECS/httpd.spec	2024-11-21 21:22:32.427194003 +0000
+++ b/SPECS/httpd.spec	2024-11-21 21:26:12.736046008 +0000
@@ -103,6 +103,8 @@
 # https://bugzilla.redhat.com/show_bug.cgi?id=
 # Patch200: httpd-2.4.X-CVE-XXXX-YYYYY.patch
 
+Patch900: mod_proxy_http.patch
+Patch901: proxypass_nomain_flag.patch
 
 License: ASL 2.0
 BuildRequires: gcc, autoconf, pkgconfig, findutils, xmlto
@@ -259,6 +261,9 @@
 %patch101 -p1 -b .full-release
 %patch102 -p1 -b .r1919325
 
+%patch900 -p1
+%patch901 -p1
+
 # Patch in the vendor string
 sed -i '/^#define PLATFORM/s/Unix/%{vstring}/' os/unix/os.h
 sed -i 's/@RELEASE@/%{release}/' server/core.c
