--- a/modules/proxy/mod_proxy.c 2023-02-09 03:46:15.082246669 +0000
+++ b/modules/proxy/mod_proxy.c	2023-02-09 03:53:34.021474139 +0000
@@ -959,6 +959,9 @@
                 "Unescaped URL path matched ProxyPass; ignoring unsafe nocanon");
     }
 
+    if (found && r->main == NULL && (ent->flags & PROXYPASS_NOMAIN))
+        found = NULL;
+
     if (found) {
         /* A proxy module is assigned this URL, check whether it's interested
          * in the request itself (e.g. proxy_wstunnel cares about Upgrade
@@ -2062,6 +2065,9 @@
         else if (!strcasecmp(word,"noquery")) {
             flags |= PROXYPASS_NOQUERY;
         }
+        else if (!strcasecmp(word,"nomain")) {
+            flags |= PROXYPASS_NOMAIN;
+        }
         else {
             char *val = strchr(word, '=');
             if (!val) {
--- a/modules/proxy/mod_proxy.h	2023-02-09 03:46:15.083246669 +0000
+++ b/modules/proxy/mod_proxy.h	2023-02-09 03:53:58.466475305 +0000
@@ -125,6 +125,7 @@
 #define PROXYPASS_NOCANON 0x01
 #define PROXYPASS_INTERPOLATE 0x02
 #define PROXYPASS_NOQUERY 0x04
+#define PROXYPASS_NOMAIN 0x08
 #define PROXYPASS_MAP_ENCODED 0x08
 #define PROXYPASS_MAP_SERVLET 0x18 /* + MAP_ENCODED */
 struct proxy_alias {
