--- mod_reproject.cpp	2017-02-10 00:30:19.000000000 +0000
+++ mod_reproject_patched.cpp	2017-02-10 00:28:33.000000000 +0000
@@ -810,7 +810,12 @@
     static const coord_conv_f *cyf[P_COUNT] = { same_proj, wm2lat, lat2wm };
 
     // TODO: use r->header_only to verify ETags, assuming the subrequests are faster in that mode
-    repro_conf *cfg = (repro_conf *)ap_get_module_config(r->per_dir_config, &reproject_module);
+
+    //** PATCH -- pick up a modified config if one exists in the notes for this request.
+    repro_conf *cfg = ap_get_module_config(r->request_config, &reproject_module)
+        ? (repro_conf *)ap_get_module_config(r->request_config, &reproject_module) :
+            (repro_conf *)ap_get_module_config(r->per_dir_config, &reproject_module);
+    //** END PATCH
     if (!our_request(r, cfg)) return DECLINED;
 
     apr_array_header_t *tokens = tokenize(r->pool, r->uri);
