--- overview.cpp	2014-03-28 11:18:52.932001700 -0700
+++ overview.cpp	2014-03-28 11:19:50.874795400 -0700
@@ -1969,8 +1969,8 @@
 
         /* Compute the chunck size of the source such as it will match the size of */
         /* a block of the overview */
-        int nFullResXChunk = (nDstBlockXSize * nSrcWidth) / nDstWidth;
-        int nFullResYChunk = (nDstBlockYSize * nSrcHeight) / nDstHeight;
+        int nFullResXChunk = 0.5 + double(nDstBlockXSize * nSrcWidth) / nDstWidth;
+        int nFullResYChunk = 0.5 + double(nDstBlockYSize * nSrcHeight) / nDstHeight;
 
         void** papaChunk = (void**) CPLMalloc(nBands * sizeof(void*));
         GByte* pabyChunkNoDataMask = NULL;
