--- mapwms.c	2017-11-15 11:24:35.000000000 -0800
+++ mapwms_patched.c	2018-02-22 09:26:56.000000000 -0800
@@ -2039,6 +2039,7 @@
   char szVersionBuf[OWS_VERSION_MAXLEN];
   size_t bufferSize = 0;
   const char *pszDimensionlist=NULL;
+  const char *pszNearestValue=NULL;
 
   /* if the layer status is set to MS_DEFAULT, output a warning */
   if (lp->status == MS_DEFAULT)
@@ -2170,19 +2171,25 @@
     pszWmsTimeDefault = msOWSLookupMetadata(&(lp->metadata),  "MO",
                                             "timedefault");
 
+    
+    pszNearestValue = msOWSLookupMetadata(&(lp->metadata), "MO", "nearestvalue");
+    if (pszNearestValue == NULL) {
+      pszNearestValue = "0";
+    }
+
     if (nVersion >= OWS_1_3_0) {
       if (pszWmsTimeDefault)
-        msIO_fprintf(stdout, "        <Dimension name=\"time\" units=\"ISO8601\" default=\"%s\" nearestValue=\"0\">%s</Dimension>\n",pszWmsTimeDefault, pszWmsTimeExtent);
+        msIO_fprintf(stdout, "        <Dimension name=\"time\" units=\"ISO8601\" default=\"%s\" nearestValue=\"%s\">%s</Dimension>\n",pszWmsTimeDefault, pszNearestValue, pszWmsTimeExtent);
       else
-        msIO_fprintf(stdout, "        <Dimension name=\"time\" units=\"ISO8601\" nearestValue=\"0\">%s</Dimension>\n",pszWmsTimeExtent);
+        msIO_fprintf(stdout, "        <Dimension name=\"time\" units=\"ISO8601\" nearestValue=\"%s\">%s</Dimension>\n", pszNearestValue, pszWmsTimeExtent);
     }
 
     else {
       msIO_fprintf(stdout, "        <Dimension name=\"time\" units=\"ISO8601\"/>\n");
       if (pszWmsTimeDefault)
-        msIO_fprintf(stdout, "        <Extent name=\"time\" default=\"%s\" nearestValue=\"0\">%s</Extent>\n",pszWmsTimeDefault, pszWmsTimeExtent);
+        msIO_fprintf(stdout, "        <Extent name=\"time\" default=\"%s\" nearestValue=\"%s\">%s</Extent>\n",pszWmsTimeDefault, pszNearestValue, pszWmsTimeExtent);
       else
-        msIO_fprintf(stdout, "        <Extent name=\"time\" nearestValue=\"0\">%s</Extent>\n",pszWmsTimeExtent);
+        msIO_fprintf(stdout, "        <Extent name=\"time\" nearestValue=\"%s\">%s</Extent>\n", pszNearestValue, pszWmsTimeExtent);
     }
   }
 
